applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+
+#include <stdbool.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include "cJSON.h"
+#include "cJSON_os.h"
+#include "cloud_codec.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(cloud_codec, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+struct cmd {
+	char *name;
+	union {
+		enum cloud_cmd_group group;
+		enum cloud_cmd_recipient recipient;
+		enum cloud_channel channel;
+		enum cloud_cmd_type type;
+	};
+
+	struct cmd *children;
+	size_t num_children;
+};
+
+#define CMD_ARRAY(...) ((struct cmd[]) {__VA_ARGS__})
+
+#define CMD_NEW_TYPE(_name, _type)			\
+	{ 						\
+		.name = STRINGIFY(_name),		\
+		.type = _type, 				\
+	}
+
+#define CMD_NEW_CHAN(_name, _chan, _children)	\
```

Comment:
Alignment issue?

Reply:
Also on some other lines in this macro definition

Reply:
Fixed.